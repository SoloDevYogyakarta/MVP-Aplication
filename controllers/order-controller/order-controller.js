"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __param = (this && this.__param) || function (paramIndex, decorator) {
    return function (target, key) { decorator(target, key, paramIndex); }
};
var ServicesOrderController_1;
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServicesOrderController = void 0;
const common_1 = require("@nestjs/common");
const auth_guard_1 = require("../../middleware/auth-guard/auth-guard");
const order_service_1 = require("../../services/order-service/order-service");
const order_repository_1 = require("../../repository/order-repository/order-repository");
const lodash_1 = require("lodash");
let ServicesOrderController = ServicesOrderController_1 = class ServicesOrderController {
    constructor(service, repository) {
        this.service = service;
        this.repository = repository;
        this.logger = new common_1.Logger(ServicesOrderController_1.name);
    }
    async list(req, res) {
        const result = await this.repository.findAll((0, lodash_1.omit)(req.query, ['type']), req.query.type);
        return res.status(common_1.HttpStatus.OK).json(result);
    }
    async detail(req, res) {
        const result = await this.repository.findOne(req.params.id);
        return res.status(common_1.HttpStatus.OK).json(result);
    }
    async destroy(req, res) {
        this.logger.log(ServicesOrderController_1.name);
        const result = await this.service.destroy(req.params.id);
        return res.status(result.status).json(result);
    }
};
exports.ServicesOrderController = ServicesOrderController;
__decorate([
    (0, common_1.Get)(),
    (0, common_1.HttpCode)(common_1.HttpStatus.OK),
    __param(0, (0, common_1.Req)()),
    __param(1, (0, common_1.Res)()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object, Object]),
    __metadata("design:returntype", Promise)
], ServicesOrderController.prototype, "list", null);
__decorate([
    (0, common_1.Get)(':id/services'),
    (0, common_1.HttpCode)(common_1.HttpStatus.OK),
    __param(0, (0, common_1.Req)()),
    __param(1, (0, common_1.Res)()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object, Object]),
    __metadata("design:returntype", Promise)
], ServicesOrderController.prototype, "detail", null);
__decorate([
    (0, common_1.UseGuards)(auth_guard_1.AuthGuard),
    (0, common_1.Delete)(':id'),
    (0, common_1.HttpCode)(common_1.HttpStatus.OK),
    __param(0, (0, common_1.Req)()),
    __param(1, (0, common_1.Res)()),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object, Object]),
    __metadata("design:returntype", Promise)
], ServicesOrderController.prototype, "destroy", null);
exports.ServicesOrderController = ServicesOrderController = ServicesOrderController_1 = __decorate([
    (0, common_1.Controller)('order'),
    __metadata("design:paramtypes", [order_service_1.OrderService,
        order_repository_1.OrderRepository])
], ServicesOrderController);
//# sourceMappingURL=order-controller.js.map