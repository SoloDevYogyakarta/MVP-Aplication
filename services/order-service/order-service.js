"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrderService = void 0;
const common_1 = require("@nestjs/common");
const system_1 = require("../../utils/system/system");
const files_entity_1 = require("../../database/entities/services/files-entity/files-entity");
const history_entity_1 = require("../../database/entities/services/history-entity/history-entity");
const order_entity_1 = require("../../database/entities/services/order-entity/order-entity");
let OrderService = class OrderService {
    async destroy(id) {
        const findOne = await order_entity_1.orderEntity.findOne({ where: { id } });
        if (!findOne) {
            throw new common_1.HttpException({
                status: common_1.HttpStatus.BAD_REQUEST,
                message: 'Order not found',
            }, common_1.HttpStatus.BAD_REQUEST);
        }
        const history = await history_entity_1.historyEntity.findAll({
            where: { order_id: findOne.id },
        });
        for (const values of history) {
            values?.destroy();
        }
        const files = await files_entity_1.fileEntity.findAll({ where: { order_id: findOne.id } });
        for (const file of files) {
            if (file.filepath) {
                try {
                    (0, system_1.removepath)(`../..${file.filepath}`);
                }
                catch (err) {
                }
            }
            file?.destroy();
        }
        findOne.destroy();
        return { status: common_1.HttpStatus.OK, message: 'Order has been delete' };
    }
};
exports.OrderService = OrderService;
exports.OrderService = OrderService = __decorate([
    (0, common_1.Injectable)()
], OrderService);
//# sourceMappingURL=order-service.js.map