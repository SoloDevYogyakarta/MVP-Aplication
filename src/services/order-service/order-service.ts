import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
import { removepath } from '../../utils/system/system';
import { fileEntity } from '../../database/entities/services/files-entity/files-entity';
import { historyEntity } from '../../database/entities/services/history-entity/history-entity';
import { orderEntity } from '../../database/entities/services/order-entity/order-entity';

@Injectable()
export class OrderService {
  async destroy(id: number) {
    const findOne = await orderEntity.findOne({ where: { id } });
    if (!findOne) {
      throw new HttpException(
        {
          status: HttpStatus.BAD_REQUEST,
          message: 'Order not found',
        },
        HttpStatus.BAD_REQUEST,
      );
    }
    const history = await historyEntity.findAll({
      where: { order_id: findOne.id },
    });
    for (const values of history) {
      values?.destroy();
    }
    const files = await fileEntity.findAll({ where: { order_id: findOne.id } });
    for (const file of files) {
      if (file.filepath) {
        try {
          removepath(`../..${file.filepath}`);
        } catch (err) {
          // empty
        }
      }
      file?.destroy();
    }
    findOne.destroy();
    return { status: HttpStatus.OK, message: 'Order has been delete' };
  }
}
