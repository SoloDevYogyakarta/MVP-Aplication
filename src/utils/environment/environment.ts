import { config } from 'dotenv';

config();

export const environment: NodeJS.ProcessEnv = process.env;
