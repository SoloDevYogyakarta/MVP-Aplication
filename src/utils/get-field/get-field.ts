import { GetFieldType } from 'src/types/get-field.type';
import { readpath } from '../system/system';

export function getField(filename?: GetFieldType) {
  try {
    if (filename === 'fake') {
      return readpath(`../../database/dataTxt/${filename}.txt`);
    }
    return JSON.parse(readpath(`../../database/dataTxt/${filename}.txt`));
  } catch (err) {
    throw new Error(err);
  }
}
