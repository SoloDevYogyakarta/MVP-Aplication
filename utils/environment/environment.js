"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.environment = void 0;
const dotenv_1 = require("dotenv");
(0, dotenv_1.config)();
exports.environment = process.env;
//# sourceMappingURL=environment.js.map