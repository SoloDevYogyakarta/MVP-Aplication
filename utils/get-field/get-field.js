"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getField = void 0;
const system_1 = require("../system/system");
function getField(filename) {
    try {
        const inc = ['token', 'fake'];
        if (inc.includes(filename)) {
            return (0, system_1.readpath)(`../../database/dataTxt/${filename}.txt`);
        }
        return JSON.parse((0, system_1.readpath)(`../../database/dataTxt/${filename}.txt`));
    }
    catch (err) {
        throw new Error(err);
    }
}
exports.getField = getField;
//# sourceMappingURL=get-field.js.map