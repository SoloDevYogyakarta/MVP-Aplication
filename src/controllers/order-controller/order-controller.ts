import {
  Controller,
  Delete,
  HttpCode,
  HttpStatus,
  Logger,
  Req,
  Res,
  UseGuards,
} from '@nestjs/common';
import { Response } from 'express';
import { CustomRequest } from '../../types/custom-request.type';
import { AuthGuard } from '../../middleware/auth-guard/auth-guard';
import { OrderService } from '../../services/order-service/order-service';

@Controller('order')
export class ServicesOrderController {
  private readonly logger = new Logger(ServicesOrderController.name);

  constructor(private readonly service: OrderService) {}

  @UseGuards(AuthGuard)
  @Delete(':id')
  @HttpCode(HttpStatus.OK)
  async destroy(@Req() req: CustomRequest, @Res() res: Response) {
    this.logger.log(ServicesOrderController.name);
    const result = await this.service.destroy(req.params.id);
    return res.status(result.status).json(result);
  }
}
