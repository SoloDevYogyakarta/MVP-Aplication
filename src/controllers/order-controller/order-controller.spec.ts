import { HttpStatus, INestApplication } from '@nestjs/common';
import { JwtModule } from '@nestjs/jwt';
import { Test } from '@nestjs/testing';
import { getfield } from '../../utils/get-field/get-field';
import supertest from 'supertest';
import { environment } from '../../utils/environment/environment';
import { ServicesOrderController } from './order-controller';
import { OrderService } from '../../services/order-service/order-service';

describe('ServicesOrderController', () => {
  let public_id!: string;
  let token!: string;
  let app: INestApplication;

  beforeEach(async () => {
    const moduleRef = await Test.createTestingModule({
      imports: [
        JwtModule.register({
          secret: environment['SECRET'],
          signOptions: {
            expiresIn: '7d',
          },
        }),
      ],
      controllers: [ServicesOrderController],
      providers: [OrderService],
    }).compile();

    app = moduleRef.createNestApplication();
    await app.init();
  });

  it('should to be defined', () => expect(app).toBeDefined());

  it('render correctly', () => expect(app.getHttpServer()).toMatchSnapshot());

  try {
    token = getfield('token');
  } catch (err) {
    // empty
  }

  try {
    public_id = getfield('order-http-entity').public_id;
  } catch (err) {
    // empty
  }

  if (public_id) {
    it('http::order destroy', async () => {
      await supertest(app.getHttpServer())
        .delete(`/services/order/${public_id}`)
        .set('content-type', 'application/json')
        .set('Authorization', `Bearer ${token}`)
        .expect(HttpStatus.OK)
        .then((res) =>
          expect(res.body).toEqual({
            status: HttpStatus.OK,
            message: 'Order has been delete',
          }),
        );
    });

    it('http::order invalid destroy', async () => {
      await supertest(app.getHttpServer())
        .delete(`/services/order/dwqdqwd`)
        .set('content-type', 'application/json')
        .set('Authorization', `Bearer ${token}`)
        .expect(HttpStatus.BAD_REQUEST)
        .then((res) =>
          expect(res.body).toEqual({
            status: HttpStatus.BAD_REQUEST,
            message: 'Order not found',
          }),
        );
    });
  }
});
