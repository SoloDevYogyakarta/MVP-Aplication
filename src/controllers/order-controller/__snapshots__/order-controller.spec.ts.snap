// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServicesOrderController render correctly 1`] = `
Server {
  "_connections": 0,
  "_events": {
    "connection": [Function],
    "request": [Function],
  },
  "_eventsCount": 2,
  "_handle": null,
  "_maxListeners": undefined,
  "_unref": false,
  "_usingWorkers": false,
  "_workers": [],
  "allowHalfOpen": true,
  "connectionsCheckingInterval": 30000,
  "headersTimeout": 60000,
  "highWaterMark": 16384,
  "httpAllowHalfOpen": false,
  "insecureHTTPParser": undefined,
  "joinDuplicateHeaders": undefined,
  "keepAlive": false,
  "keepAliveInitialDelay": 0,
  "keepAliveTimeout": 5000,
  "maxHeaderSize": undefined,
  "maxHeadersCount": null,
  "maxRequestsPerSocket": 0,
  "noDelay": true,
  "pauseOnConnect": false,
  "rejectNonStandardBodyWrites": false,
  "requestTimeout": 300000,
  "timeout": 0,
  Symbol(IncomingMessage): [Function],
  Symbol(ServerResponse): [Function],
  Symbol(kCapture): false,
  Symbol(async_id_symbol): -1,
  Symbol(http.server.connections): ConnectionsList {},
  Symbol(http.server.connectionsCheckingInterval): Timeout {
    "_destroyed": false,
    "_idleNext": Timeout {
      "_destroyed": false,
      "_idleNext": TimersList {
        "_idleNext": [Circular],
        "_idlePrev": [Circular],
        "expiry": 40701,
        "id": -9007199254740968,
        "msecs": 30000,
        "priorityQueuePosition": 3,
      },
      "_idlePrev": [Circular],
      "_idleStart": 10701,
      "_idleTimeout": 30000,
      "_onTimeout": [Function],
      "_repeat": 30000,
      "_timerArgs": undefined,
      Symbol(refed): false,
      Symbol(kHasPrimitive): false,
      Symbol(asyncId): 169,
      Symbol(triggerId): 0,
    },
    "_idlePrev": TimersList {
      "_idleNext": [Circular],
      "_idlePrev": Timeout {
        "_destroyed": false,
        "_idleNext": [Circular],
        "_idlePrev": [Circular],
        "_idleStart": 10701,
        "_idleTimeout": 30000,
        "_onTimeout": [Function],
        "_repeat": 30000,
        "_timerArgs": undefined,
        Symbol(refed): false,
        Symbol(kHasPrimitive): false,
        Symbol(asyncId): 169,
        Symbol(triggerId): 0,
      },
      "expiry": 40701,
      "id": -9007199254740968,
      "msecs": 30000,
      "priorityQueuePosition": 3,
    },
    "_idleStart": 10716,
    "_idleTimeout": 30000,
    "_onTimeout": [Function],
    "_repeat": 30000,
    "_timerArgs": undefined,
    Symbol(refed): false,
    Symbol(kHasPrimitive): false,
    Symbol(asyncId): 173,
    Symbol(triggerId): 0,
  },
  Symbol(kUniqueHeaders): null,
}
`;
